### DISCUSSION TITLE
Warnings of missing return statements and linker issues while compiling C++ examples

### DISCUSSION THREAD

**Discussion by Shane-Xue** [webui-dev/webui#530] [Q&A]
Hello,

Since I am new to webui I decide to start with the examples in ./webui/examples/C++. I started with minimal and got various issues:

1.  Lack of return statement in wrapper functions: `bool webui::window::event::show_client` and `std::string_view webui::window::start_server`;
2. undefined reference to `__imp_CoTaskMemFree'` in linker. (`L:/Code/webui/examples/C++/serve_a_folder//../../..//dist\libwebui-2-static.a(webui.o):webui.c:(.text+0x5ac): undefined reference to __imp_CoTaskMemFree'`)

Adding a return statement to the prior two functions (clearly) will resolve the warning, but I am not familiar with the latter. There is a call to CoTaskMemFree in webui.c though.

I am using a Windows 11 platform combined with mingw32-make with gcc 14.2.0. I used the official makefiles that came with the source code.

Compiling the C examples worked fine. What might I do so find and solve my problem?

**Comment by AlbertShown** [comment]
Thank you @Shane-Xue for reporting that.

All C++ GNU Makefiles are updated now (https://github.com/webui-dev/webui/commit/2a406f5276f854feba567b55aae154df0730ed30).
Please clone, and re-build.

_Tested using `x86_64-14.2.0-release-win32-seh-msvcrt-rt_v12-rev0`_
